runtime/src/accounts_index.rs

```diff
@@ -1231,8 +1235,12 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
                 }
                 if let Some(mut w_account_entry) = account_entry {
                     w_account_entry.update(slot, account_info, &mut _reclaims);
+                    true
+                } else {
+                    false
```

Comment:
Just to make sure I'm understanding the reasoning correctly, for this:
```
if let Some(mut w_account_entry) = account_entry {
     true
}
```

Combined with:
```
 dirty_pubkeys.retain(|_k| *dirty_pubkey_mask_iter.next().unwrap());
```
means if there was a previous entry for this pubkey we add it to the dirty pubkey list, but if there was no previous entry, we skip adding it to the dirty pubkey list. 

Is the reasoning that if this pubkey is only updated once, then it doesn't have a later update which renders the first update obsolete, which means it doesn't need to be cleaned? 

The one case where we might still need this is if the account is zero lamports so that it can be added to the zero-lamport list here: https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_db.rs#L1654-L1658

Reply:
Can we also add a note/issue/todo to add a test for the `generate_index() -> clean_accounts()` flow to test some of these corner cases? This logic is getting complicated enough where I think it's warranted

Reply:
> Is the reasoning that if this pubkey is only updated once, then it doesn't have a later update which renders the first update obsolete, which means it doesn't need to be cleaned?

Yes, exactly. If there is just 1 update for a given key, there is no cleaning that could be done unless that key is 0-lamport and that is happening through a different path.